### DISCUSSION TITLE
How to run example“Horizontal_1D_model”？

### DISCUSSION THREAD

**Discussion by CUGWH** [acroucher/PyTOUGH#52] [Q&A] [answered]
Hello！
I'm very new to TOUGH2 and I started using PyTOUGH.
I tried running a sample code "Horizontal 1D Model" in PyTOUGH's wiki page,
but I got following error message:
Traceback (most recent call last):
  File "E:\OneDrive - China University of Geosciences(Wuhan)\PyTOUGH_project\Horizontal_1D_model\Horizontal_1D_model.py", line 75, in <module>
    lst = t2listing('horiz1D.listing')
  File "D:\ProgramData\Anaconda3\envs\PyTOUGH_v157\lib\site-packages\pytough-1.5.7-py3.8.egg\t2listing.py", line 177, in __init__
Exception: Could not detect simulator type.
Does anyone know how to use method of "t2.data.run()" ??

Hao Wei

**Comment by acroucher** [comment, marked as answer]
hi,

Did you have a look at the listing file 'horiz1D.listing' to see what is in it?

Usually if PyTOUGH cannot detect the simulator type of the listing file, it means the simulation didn't run properly. In that case you may want to try running the simulation manually first before running it using `t2data.run()`.  That way you can see the output and find out where things are going wrong.

**Reply by CUGWH** [reply]
ok. Thanks for your answer!
